File: src/main/java/railgunDownloaderV4/components/Youtube.java
/*
 * RAILGUN DOWNLOADER - VERSION 4.0.0
 * Youtube.java
 */
package railgunDownloaderV4.components;
import javax.swing.*;
import java.awt.*;
import java.util.Objects;
import railgunDownloaderV4.Application;
import railgunDownloaderV4.components.youtubeUI.YoutubeUI;

public class Youtube {
    private final Application AppScene;
    private  final YoutubeUI UIContext;

    public  Youtube(Application AppScene) {
        this.AppScene = AppScene;
        UIContext = new YoutubeUI(AppScene);
    }

    public  void SetYoutubeButton(JButton YoutubeButton) {
        YoutubeButton.setSize(50, 50);
        YoutubeButton.setBorderPainted(false);
        YoutubeButton.setBorder(null);
        YoutubeButton.setContentAreaFilled(false);
        YoutubeButton.setFocusPainted(false);

        Image ButtonIcon = Toolkit.getDefaultToolkit().getImage(
                Objects.requireNonNull(
                        getClass().getResource("/YoutubeIcon.png")
                )
        );
        YoutubeButton.setIcon(new ImageIcon(
                ButtonIcon.getScaledInstance(50, 50, Image.SCALE_FAST)
        ));

        YoutubeButton.setToolTipText("Download video from Youtube");

        OpenYoutubeUI(YoutubeButton);
        AppScene.App.add(YoutubeButton);
    }

    private void OpenYoutubeUI(JButton YoutubeButton) {
        SwingUtilities.invokeLater(() ->
                YoutubeButton.addActionListener(actionEvent -> {
                UIContext.ShowYoutubeUI(true);
                AppScene.App.setVisible(false);
        }));
    }
}

File: src/main/kotlin/railgunDownloaderV4/components/Youtube.kt
/*
 * RAILGUN DOWNLOADER - VERSION 4.0.0
 * File: Youtube.kt
 * This project is licensed GPL-3.0 License.
 * LICENSE: https://www.gnu.org/licenses/gpl-3.0.html
 * Contribution: Reim
 */
package railgunDownloaderV4.components

import railgunDownloaderV4.Application
import railgunDownloaderV4.components.youtubeUI.YoutubeUI
import java.awt.Image
import java.awt.Toolkit
import java.util.*
import javax.swing.ImageIcon
import javax.swing.JButton
import javax.swing.SwingUtilities

class Youtube(private val appScene: Application) {
    private val uiContext = YoutubeUI(appScene)

    fun setYoutubeButton(youtubeButton: JButton) {
        youtubeButton.setSize(50, 50)
        youtubeButton.isBorderPainted = false
        youtubeButton.border = null
        youtubeButton.isContentAreaFilled = false
        youtubeButton.isFocusPainted = false

        val buttonIcon = Toolkit.getDefaultToolkit().getImage(
            Objects.requireNonNull(
                javaClass.getResource("/YoutubeIcon.png")
            )
        )
        youtubeButton.icon = ImageIcon(
            buttonIcon.getScaledInstance(50, 50, Image.SCALE_FAST)
        )

        youtubeButton.toolTipText = "Download video from Youtube"

        openYoutubeUI(youtubeButton)
        appScene.app.add(youtubeButton)
    }

    private fun openYoutubeUI(youtubeUI: JButton) {
        SwingUtilities.invokeLater {
            youtubeUI.addActionListener {
                uiContext.showYoutubeUI(true)
                appScene.app.isVisible = false
            }
        }
    }
}
